app/javascript/App/components/NewsfeedPage/CoinList.js

```diff
@@ -1,29 +1,58 @@
-import React, { Fragment } from 'react'
+import React, { Component, Fragment } from 'react'
 import CoinListHeader from './CoinListHeader'
 import CoinListItem from './CoinListItem'
 import LoadingIndicator from '../LoadingIndicator'
+import coinSearchProvider from '../../containers/coinSearch'
 
-export default (props) => {
-  const { isLoading, setActiveEntity, currentUI, isWatching } = props
-  const onClick = (coin) =>
-    setActiveEntity({
+class CoinList extends Component {
+  onClickCoin = (coin) =>
+    this.props.setActiveEntity({
       type: 'coin',
       id: coin.get('id'),
       label: coin.get('name')
     })
-  let { coins } = props
-  if (currentUI('watchingOnly')) {
-    coins = props.coins.filter((coin) => isWatching(coin.get('id')))
+  onClickNewCoin = (coin) => {
+    this.onClickCoin(coin)
+    this.props.toggleUI('coinSearch')
+    this.props.clearSearch()
+  }
+  render() {
+    const { isLoading, currentUI, isWatching, searchedCoins } = this.props
+    let { coins } = this.props
+    if (currentUI('watchingOnly')) {
+      coins = coins.filter((coin) => isWatching(coin.get('id')))
+    }
+    return (
+      <Fragment>
+        <CoinListHeader {...this.props} />
+        <div className="flex-auto relative overflow-y-auto">
+          {isLoading('coins') && (
+            <LoadingIndicator className="overlay bg-white-70" />
+          )}
+          {currentUI('coinSearch') &&
+            searchedCoins.size > 0 && (
```

Comment:
any values greater than zero will always evaluate to true.

Reply:
I think having `> 0` is a bit more human-friendly